enterprise/wsproxy/wsproxy_test.go

```diff
@@ -498,3 +600,86 @@ func TestWorkspaceProxyWorkspaceApps(t *testing.T) {
 		}
 	})
 }
+
+// createDERPClient creates a DERP client and spawns a goroutine that reads from
+// the client and sends the received packets to a channel.
+func createDERPClient(t *testing.T, name string, derpURL string) (*derphttp.Client, <-chan derp.ReceivedPacket) {
+	t.Helper()
+
+	client, err := derphttp.NewClient(key.NewNode(), derpURL, func(format string, args ...any) {
+		t.Logf(name+": "+format, args...)
+	})
+	require.NoError(t, err, "create client")
+	t.Cleanup(func() {
+		_ = client.Close()
+	})
+	err = client.Connect(testutil.Context(t, testutil.WaitLong))
+	require.NoError(t, err, "connect to DERP server")
+
+	ch := make(chan derp.ReceivedPacket, 1)
+	done := make(chan struct{})
+	go func() {
+		defer close(ch)
+		defer close(done)
+		for {
+			msg, err := client.Recv()
+			if err != nil {
+				t.Logf("Recv error: %v", err)
+				return
+			}
+			switch msg := msg.(type) {
+			case derp.ReceivedPacket:
+				ch <- msg
+				return
+			default:
+				// We don't care about other messages.
+			}
+		}
+	}()
+	t.Cleanup(func() {
+		<-done
```

Comment:
I think you're probably doing this to make sure we don't leak the receive packet goroutine.  For that, `goleak` is the way forward.  Blocking here doesn't _prevent_ it from leaking; it just blocks test cleanup, meaning that a leak will cause the whole test suite to hang for 10 minutes until it's killed.  `goleak` will return in a few seconds.